### PR TITLE
Fix part of #40 & #42: Generalisation mobile devices - Content View - Portrait

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

This PR finishes the highfi work of `ContentView-Portrait Mode` for mobile devices only.

Note, the strategy that I have followed for margins is that we we introduce only top-start-end margins and bottom margin won't be introduced (as bottom margin for item1 can be treated as top margin for item-2).

Note: When you run the app, it might seem that the text size is small, so to fix that, go to `Options -> StoryTextSize` and set it to medium.

## Reference: 
exploration_portrait
question_portrait

## Screenshots
Screenshot_1595713286

Screenshot_1595713260
 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
+package org.oppia.app.databinding
+
+import android.view.View
+import android.view.ViewGroup.MarginLayoutParams
+import androidx.databinding.BindingAdapter
+
+/** Used to set a margin for exploration-split view. */
```
Comment (MohamedMedhat1998) [inline app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt]:
```suggestion
/** Used to set a margin for exploration split-view. */
```

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  if (isApplicable && view.layoutParams is MarginLayoutParams) {
+    val params = view.layoutParams as MarginLayoutParams
+    params.setMargins(
+      marginStart.toInt(),
+      marginTop.toInt(),
+      marginEnd.toInt(),
+      marginBottom.toInt()
+    )
+    view.requestLayout()
+  }
```
Comment (MohamedMedhat1998) [inline app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt]: I suggest moving this implementation to a new function and reuse it in all the other functions.  
Example:  
```
fun setValues(
  view: View,
  isApplicable: Boolean,
  marginStart: Float,
  marginTop: Float,
  marginEnd: Float,
  marginBottom: Float
) {
  if (isApplicable && view.layoutParams is MarginLayoutParams) {
    val params = view.layoutParams as MarginLayoutParams
    params.setMargins(
      marginStart.toInt(),
      marginTop.toInt(),
      marginEnd.toInt(),
      marginBottom.toInt()
    )
    view.requestLayout()
  }
}
```
```
fun setExplorationSplitViewMargin(
  view: View,
  isApplicable: Boolean,
  marginStart: Float,
  marginTop: Float,
  marginEnd: Float,
  marginBottom: Float
) {
  setValues(view, isApplicable, marginStart, marginTop, marginEnd, marginBottom)
}
```

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+/** Used to set a margin for question-split view. */
```
Comment (MohamedMedhat1998) [inline app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt]:
```suggestion
/** Used to set a margin for question view. */
```

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      marginStart.toInt(),
+      marginTop.toInt(),
+      marginEnd.toInt(),
+      marginBottom.toInt()
+    )
+    view.requestLayout()
+  }
+}
+
+/** Used to set a margin for question view. */
```
Comment (MohamedMedhat1998) [inline app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt]:
```suggestion
/** Used to set a margin for question split-view. */
```

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt [hunk 5]
```diff
@@ -0,0 +1,112 @@
+package org.oppia.app.databinding
+
+import android.view.View
+import androidx.databinding.BindingAdapter
+
+/** Used to set a padding for exploration-split view. */
```
Comment (MohamedMedhat1998) [inline app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]:
```suggestion
/** Used to set a padding for exploration split-view. */
```

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+  paddingBottom: Float
+) {
+  if (isApplicable) {
+    view.setPadding(
+      paddingStart.toInt(),
+      paddingTop.toInt(),
+      paddingEnd.toInt(),
+      paddingBottom.toInt()
+    )
+  }
```
Comment (MohamedMedhat1998) [inline app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]: Suggesting moving the implementation to a new function as described before

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+/** Used to set a padding for question-split view. */
```
Comment (MohamedMedhat1998) [inline app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]:
```suggestion
/** Used to set a padding for question view. */
```

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    view.setPadding(
+      paddingStart.toInt(),
+      paddingTop.toInt(),
+      paddingEnd.toInt(),
+      paddingBottom.toInt()
+    )
+  }
+}
+
+/** Used to set a padding for question view. */
```
Comment (MohamedMedhat1998) [inline app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]:
```suggestion
/** Used to set a padding for question split-view. */
```

**Review by MohamedMedhat1998 — CHANGES_REQUESTED** [review]
LGTM but there are some suggestions

**Comment by MohamedMedhat1998** [comment]
> exploration_portrait

The margins in this reference is different from the mocks: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/9ea1d153-06e3-434f-9bf7-d582d48583b1/specs/

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt (hunk 1 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt]: Done.

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt (hunk 2 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt]: Done.

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt (hunk 3 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt]: Done.

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt (hunk 4 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt]: Done.

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt (hunk 5 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]: Done.

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt (hunk 6 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]: Done.

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt (hunk 7 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]: Done.

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt (hunk 8 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]: Done.

**Comment by rt4914** [comment]
> > exploration_portrait
> 
> The margins in this reference is different from the mocks: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/9ea1d153-06e3-434f-9bf7-d582d48583b1/specs/

Actually, the mocks are inconsistent, that's why we have received this from Chantel, in an email thread.

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt (hunk 4 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt]: The kdoc is different from what the function is doing. The kdoc is `question view` and it should be `question split-view` and vice versa

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt (hunk 7 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]: Ditto

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt (hunk 8 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]: Ditto

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt (hunk 4 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerMarginBindingAdapters.kt]: Done.

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt (hunk 7 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]: Done.

File: app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt (hunk 8 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/databinding/StateAssemblerPaddingBindingAdapters.kt]: Done.

**Review by MohamedMedhat1998 — APPROVED** [review]
LGTM

**Pull request merged**